src/Datasource/ModelAwareTrait.php

```diff
@@ -1,167 +0,0 @@
- * @link          https://cakephp.org CakePHP(tm) Project
- * @since         3.0.0
- * @license       https://opensource.org/licenses/mit-license.php MIT License
- */
-namespace Cake\Datasource;
-
-use Cake\Datasource\Exception\MissingModelException;
-use Cake\Datasource\Locator\LocatorInterface;
-use UnexpectedValueException;
-
-/**
- * Provides functionality for loading table classes
- * and other repositories onto properties of the host object.
- *
- * Example users of this trait are Cake\Controller\Controller and
- * Cake\Console\Command.
- */
-trait ModelAwareTrait
-{
-    /**
-     * This object's primary model class name. Should be a plural form.
-     * CakePHP will not inflect the name.
-     *
-     * Example: For an object named 'Comments', the modelClass would be 'Comments'.
-     * Plugin classes should use `Plugin.Comments` style names to correctly load
-     * models from the correct plugin.
-     *
-     * Use empty string to not use auto-loading on this object. Null auto-detects based on
-     * controller name.
-     *
-     * @var string|null
-     */
-    protected ?string $modelClass = null;
-
-    /**
-     * A list of overridden model factory functions.
-     *
-     * @var array<callable|\Cake\Datasource\Locator\LocatorInterface>
-     */
-    protected array $_modelFactories = [];
-
-    /**
-     * The model type to use.
-     *
-     * @var string
-     */
-    protected string $_modelType = 'Table';
-
-    /**
-     * Set the modelClass property based on conventions.
-     *
-     * If the property is already set it will not be overwritten
-     *
-     * @param string $name Class name.
-     * @return void
-     */
-    protected function _setModelClass(string $name): void
-    {
-        if ($this->modelClass === null) {
-            $this->modelClass = $name;
-        }
-    }
-
-    /**
-     * Loads and constructs repository objects required by this object
-     *
-     * Typically used to load ORM Table objects as required. Can
-     * also be used to load other types of repository objects your application uses.
-     *
-     * If a repository provider does not return an object a MissingModelException will
-     * be thrown.
-     *
-     * @param string|null $modelClass Name of model class to load. Defaults to $this->modelClass.
-     *  The name can be an alias like `'Post'` or FQCN like `App\Model\Table\PostsTable::class`.
-     * @param string|null $modelType The type of repository to load. Defaults to the getModelType() value.
-     * @return \Cake\Datasource\RepositoryInterface The model instance created.
-     * @throws \Cake\Datasource\Exception\MissingModelException If the model class cannot be found.
-     * @throws \UnexpectedValueException If $modelClass argument is not provided
-     *   and ModelAwareTrait::$modelClass property value is empty.
-     */
-    public function loadModel(?string $modelClass = null, ?string $modelType = null): RepositoryInterface
```

Comment:
We will need to add a deprecation for this in 4.3

Reply:
That would require backporting this change to 4.3 which I was hoping to avoid :slightly_smiling_face:. Given how commonly `loadModel()` and `$modelClass` would be used in apps and plugins, the users could be quite annoyed by it's deprecation. People were quite unamused when we started showing deprecation errors for `TableRegistry`'s methods in a minor release.

So I was hoping we could do a clean switch in 5.0 directly and provide rector rules to replace use of `loadModel()` with `getTable()` and `$modelClass` with `$defaultTable`.

Reply:
I'm cautious of including a hard break with no deprecation as we can't say 'fix all your deprecations and you are pretty much done' anymore. Could we include `getTable()` in 4.x and update the documentation? Perhaps we deprecate `loadModel()` in a future 4.x release which doesn't have as many deprecations as 4.3 will contain.

Reply:
Sure, we can do. I'll make a PR to add `getTable()` to 4.3.